parladata/admin/memberships.py

```diff
@@ -8,9 +10,68 @@
 from import_export.admin import ImportExportModelAdmin
 
 
+
+
+
+class PersonMembershipForm(forms.ModelForm):
+    class Meta:
+        model = PersonMembership
+        exclude = []
+
+    def clean(self):
+        start_time = self.cleaned_data.get('start_time')
+        end_time = self.cleaned_data.get('end_time')
+        member = self.cleaned_data.get('member')
+        organization = self.cleaned_data.get('organization')
+        memberships = PersonMembership.objects.filter(
+            member=member,
+            organization=organization,
+        )
+        print('CLEAN')
+        # check for overlapping memberships
+        if end_time:
+            # check if new membership ends inside existing membership
+            if memberships.filter(
+                    Q(start_time__lte=end_time) |
+                    Q(start_time=None),
+                    Q(end_time__gte=end_time) |
+                    Q(end_time=None)
+                ).exclude(
+                    pk=self.instance.pk
+                ).exists():
+                raise forms.ValidationError("This membership ends inside existing membership")
+
+            if start_time:
+                # check if new membership starts before and ends after existing membership
+                if memberships.filter(
+                        start_time__lte=start_time,
+                        end_time__gte=end_time
+                    ).exclude(
+                        pk=self.instance.pk
+                    ).exists():
+                    raise forms.ValidationError("Membership is around existing membership")
```

Comment:
I suggest: "This membership begins while a previously existing membership for this person in this organisation is active. Memberships (by same people in same ogranisations) should not overlap, please fix the start and end times."